docs/proposals/05-etcd-operator-tasks.md

```diff
@@ -0,0 +1,365 @@
+* All tasks have to adhere some prerequisites(a.k.a `pre-conditions`) which will be necessary to execute the task.
+* Authors propose to define pre-conditions for each task, which must be met for the task to be executed.
+* If multiple tasks are invoked simultaneously or in `pending` state, then they will be executed in a First-In-First-Out (FIFO) manner.
+
+> Note: Dependent ordering among tasks will be addresed later which may enable concurrent execution of tasks.
+
+#### Deletion
+
+`Etcd-druid` will ensure that it garbage collects the custom resources and any other k8s resources created to realise the task as per `.spec.ttlSecondsAfterFinished` defined (or default time) in the [spec](#spec) of custom resource.
+
+### Use Cases
+
+#### Recovery from permanent quorum loss
+
+Currently identification and recovery from permament quorum loss is done manaually. The current proposal keeps the identification as manual, requiring an human operator to intervene and confirm that there is indeed a permanent quorum loss and automatic recovery is not possible. Once that is established then the authors now wish to automate the recovery which is currently a [multi-step process](https://github.com/gardener/etcd-druid/blob/master/docs/operations/recovery-from-permanent-quorum-loss-in-etcd-cluster.md) and needs to be done carefully. Automation would ensure that we eliminate any errors from an operator.
+
+##### Task Config
+
+We do not need any config for this task. When creating an instance of `EtcdOperatorTask` for this scenario, `.spec.config` will be set to nil (unset).
+
+##### Pre-Conditions
+
+* There should be a quorum loss in a multi-member etcd cluster. For a single-member etcd cluster, invoking this task is unnecessary as the restoration of the single member is automatically handled by the backup-restore process.
+* There should not already be a permanent-quorum-loss-recovery-task running for the same etcd cluster.
+
+#### Trigger on-demand snapshot compaction
+
+`etcd-druid` provides a configurable [etcd-events-threshold](https://github.com/gardener/etcd-druid/blob/master/docs/proposals/02-snapshot-compaction.md#druid-flags). If and when this threshold is breached then only a [snapshot compaction](https://github.com/gardener/etcd-druid/blob/master/docs/proposals/02-snapshot-compaction.md) is triggered for etcd cluster. However, there are scenarios where an ad-hoc snapshot compaction is required.
+
+##### Possible scenarios
+
+* Full snapshot taken via compaction job is better as compared to full snapshot taken directly from etcd cluster as full snapshot taken via snapshot compaction is taken after [defragmentation](https://etcd.io/docs/v3.2/op-guide/maintenance/#defragmentation) called on etcd, hence this full snapshot is much more compacted, hence less in size.
+* Full snapshot triggered on etcd cluster might cause load on running etcd cluster but full snapshot taken by compaction job is developed separately with zero effect on running cluster.
```

Comment:
```suggestion
* Full snapshot triggered on etcd cluster might cause load on running etcd cluster but full snapshot taken by compaction job is done separately with zero effect on the running cluster.
```

Reply:
I would like to stick with `developed` word here as it make more sense since snapshot compaction is multi-step process then only a full snapshot is taken, hence I went with word `developed`.

Reply:
`developed` is an incorrect word choice here. What you wish to say here is compaction job is `deployed` or `executed` in a dedicated pod started when a k8s `Job` is scheduled.  There is nothing `developed` separately.

Reply:
> There is nothing developed separately.

`developed` is used in context of full snapshot taken by compaction sub-command which starts embedded etcd and so on ... so, IMO it's `develop`. I can re-phrase the whole sentence with something like this:

> Full snapshot triggered on etcd cluster might cause load on running etcd cluster but the full snapshot taken by the compaction job is designed to run separately, thereby having no impact on the operational cluster.

Reply:
then instead of `developed` it should be `created`. But your rephrase is better and you can stick to that as that is much more clearer.